### PR TITLE
[5.2] migrate expired delayed jobs but not expired reserved jobs when expire is null

### PR DESCRIPTION
**Pull request by halaei**
`migrateAllExpiredJobs()` can get deprecated.

### CODE REVIEW

**Comment by halaei** [comment]
Issue: #12595

File: src/Illuminate/Queue/RedisQueue.php [hunk 1]
```diff
@@ -129,8 +129,9 @@ public function pop($queue = null)
 
         $queue = $this->getQueue($queue);
 
+        $this->migrateExpiredJobs($queue.':delayed', $queue);
         if (! is_null($this->expire)) {
```
Comment (kevinsimard) [inline src/Illuminate/Queue/RedisQueue.php]: Would probably add a newline between the $this->migrateExpiredJobs(... and the if condition no?

**Comment by taylorotwell** [comment]
Double check with a test.

**Comment by halaei** [comment]
Having the following configuration for Redis queue, no delayed job will ever get extracted from the delayed queue:
```php
        'redis' => [
            'driver'     => 'redis',
            'connection' => 'default',
            'queue'      => 'default',
            'expire'     => null,
        ],
```

**Comment by halaei** [comment]
Should this be reopened?

**Pull request closed without merging**